src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.merge(
+                transaction.getFruit(),
+                -transaction.getQuantity(),
+                Integer::sum
+        );
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. If the balance is negative after applying the transaction, you should throw a RuntimeException to prevent incorrect data in the Storage.